components/script/script_thread.rs

```diff
@@ -1035,6 +1037,16 @@ impl ScriptThread {
                 image_cache_port.add();
             }
 
+            // Notify the background-hang-monitor of all the pipeline waiting for an event.
+            let waiting_components: Vec<MonitoredComponentType> = self
+                .documents
+                .borrow()
+                .iter()
+                .map(|id, _| MonitoredComponentType::Script(id.clone()))
+                .collect();
+            let msg = MonitoredComponentMsg::NotifyWait(waiting_components);
+            let _ = self.script_sender.send((id.clone(), ScriptMsg::ForwardToBackgroundHangMonitor(msg)));
```

Comment:
Yes that's a good idea(I looked at using the `TopLevelBrowsingContextId`, and since we introduced auxiliaries  a script-thread could also manage multiple of those. 

I can add a new Id to the existing `EventLoop` concept.

Reply:
An event-loop id might also not well reflect the reality, since while we indeed will have one script-thread per event-loop, we can have several layout threads. For now I have opted for this https://github.com/servo/servo/pull/21673/files#diff-8dc9268416cd4f037d50206e76c73cf9R525